### PR TITLE
feat(cb2-2941): add husky and fix linting issues

### PR DESCRIPTION
**Pull request by aaron-downing**
## Set up new Angular VTM project in code
Fixes ESLint/Prettier and adds Husky git hooks to the repo
[link to ticket number](https://jira.dvsacloud.uk/browse/CB2-2877)

## Checklist

- [X] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [X] Code and UI has been tested manually after the additional changes
- [X] PR title includes the JIRA ticket number
- [X] Squashed commits contain the JIRA ticket number
- [X] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by tom-bjss — APPROVED** [review]

File: commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      // disable it to allow release notes from semantic-release to happen when releasing
+      'footer-max-line-length': [0],
+      'body-max-line-length': [0],
+      // level: disabled to use our own rule with commitlint-plugin-function-rules plugin
+      'scope-case': [0],
+      'function-rules/scope-empty': [
+        2,
+        'always',
+        (parsed) => {
+          const ticketNumberFormat = /^cb2-\d+$/;
```
Comment (Ian-Hodges) [inline commitlint.config.js:24]: Need to change this to be the less restrictive version.
`const ticketNumberFormat = /^[a-z0-9]+-\d+$/;`

File: package.json [hunk 2]
```diff
@@ -13,15 +13,16 @@
     "build": "ng build --build-optimizer --delete-output-path --optimization --progress --vendor-chunk --aot",
     "watch": "ng build --watch --configuration development",
     "sonar-scanner": "npm run test && sonar-scanner",
-    "security-checks": "git secrets --scan && git log -p | scanrepo",
```
Comment (Ian-Hodges) [inline package.json:16]: I'd say we need to keep scanrepo until we are officially told not to use it anymore.

File: src/app/layout/footer/footer.component.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ import { Component, OnInit } from '@angular/core';
 export class FooterComponent implements OnInit {
 
   constructor() { }
-
+  /* eslint-disable @angular-eslint/no-empty-lifecycle-method*/
```
Comment (NiallBunting) [inline src/app/layout/footer/footer.component.ts:11]: Can we just delete `ngOnInit` here?

File: package.json (hunk 2 above)
Comment (tom-bjss) [reply package.json:16]: I agree

**Pull request closed without merging**